File: springboot-remoting-client/src/main/java/com/example/remoting/DnsService.java
package com.example.remoting;

import javax.jws.WebMethod;
import javax.jws.WebService;
import javax.jws.soap.SOAPBinding;
import javax.jws.soap.SOAPBinding.Style;

@WebService(targetNamespace="http://www.example.com/")
@SOAPBinding(style = Style.DOCUMENT)
public interface DnsService {
    @WebMethod
    String getName(String ip);
}

File: springboot-remoting-kotlin-server/src/main/kotlin/com/example/remoting/service/DnsService.kt
package com.example.remoting.service

import com.example.remoting.Const
import javax.jws.WebMethod
import javax.jws.WebService
import javax.jws.soap.SOAPBinding
import javax.jws.soap.SOAPBinding.Style

@WebService(targetNamespace = Const.NAMESPACE)
@SOAPBinding(style = Style.DOCUMENT)
interface DnsService {
    @WebMethod
    fun getName(ip: String): String
}